plugins/modules/git_config.py

```diff
@@ -36,12 +36,14 @@
   list_all:
     description:
       - List all settings (optionally limited to a given O(scope)).
+      - Reading git configs is deprecated and will be removed in community.general 9.0.0. Replacement module is git_config_info.
```

Comment:
Please move the deprecation to a separate PR. Features should only be deprecated a certain time after the feature replacing them has been added, to avoid unnecessary frustration of users who have to support multiple versions of the collection.

(Also removal in 9.0.0 is **way** too early.)